common/motion-dashboard.cpp

```diff
@@ -0,0 +1,146 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include "motion-dashboard.h"
+
+using namespace rs2;
+using namespace rsutils::string;
+
+motion_dashboard::motion_dashboard( std::string name )
+    : stream_dashboard( name, 30 )
+    , last_time( glfwGetTime() )
+    , x_value( 0 )
+    , y_value( 0 )
+    , z_value( 0 )
+    , n_value( 0 )
+{
+    frame_rate = MAX_FRAME_RATE;
+    clear( true );
+}
+
+void motion_dashboard::draw( ux_window & win, rect r )
+{
+    if( accel_params[curr_accel_param_position] == "X" )
+    {
+        auto x_hist = read_shared_data< std::deque< float > >( [&]() { return x_history; } );
```

Comment:
We create a copy of x_history deque, just to build a vector point by point in a loop.
Very inefficient.
Try changing `stream_dashboard::xy` to deque instead of vector and copy the data directly to it (or create a copy like here and move it to `xy`).

Reply:
> copy the data directly
Do you mean **don't** use `read_shared_data` function?

Reply:
Good question. We do need the lock.
For C++17 there is Return Value Optimization that will eliminate the copy in `read_shared_data` return.
Currently we use C++14, so it is not guaranteed for us.
Most of the benefit will probably come from changing `xy` to deque and setting all it's data at once, so lets start with that.